odoo/modules/module.py

```diff
@@ -367,6 +368,24 @@ def load_openerp_module(module_name):
         if info['post_load']:
             getattr(sys.modules[qualname], info['post_load'])()
 
+    except AttributeError as err:
+        _logger.critical("Couldn't load module %s", module_name)
+        trace = traceback.format_exc()
+        matching = re.search(r' *(\w+)(?: *: ([^ ]*))? *= *fields[.](One2many|Many2many|Many2one)(?:\[([^\]]*)\])?', trace)
+        if 'most likely due to a circular import' in trace and matching:
+            field_name = matching.group(1)
+            ttype = matching.group(3)
+            addons_class = matching.group(2) or matching.group(4)
+            addon_name = addons_class.split('.')[0] if '.' in addons_class else module_name
+            class_name = addons_class.split('.')[-1]
+            raise AttributeError(
+                str(err) + "\n"
+                "To avoid circular import for the the comodel use the annotation syntax:\n"
+                f"    {field_name}: {addon_name}.{class_name} = fields.{ttype}(...)\n"
+                "and add at the beggining of the file:\n"
+                "    from __future__ import annotations"
+                ).with_traceback(err.__traceback__) from None
```

Comment:
Please define a constant for that regular expression
```py
TYPED_FIELD_DEFINITION_RE = re.compile(r'''
    \b(?P<field_name>\w+)\s*
    (:\s*(?P<field_type>[^ ]*))?\s*
    =\s*fields\.(?P<field_class>One2many|Many2many|Many2one)
    (\[(?P<type_param>[^\]]+)\])?
''', re.VERBOSE)
```
and use more standard variable naming for regex matches (_matching_ is a verb)
```suggestion
        match = TYPED_FIELD_DEFINITION_RE.search(trace)
        if match and "most likely due to a circular import" in trace:
            field_name = match['field_name']
            field_class = match['field_class']
            field_type = match['field_type'] or match['type_param']
            if "." not in field_type:
                field_type = f"{module_name}.{field_type}"
            raise AttributeError(
                f"{err}\n"
                "To avoid circular import for the the comodel use the annotation syntax:\n"
                f"    {field_name}: {field_type} = fields.{field_class}(...)\n"
                "and add at the beggining of the file:\n"
                "    from __future__ import annotations"
            ).with_traceback(err.__traceback__) from None
```
Note that I've used named capturing groups, which greatly improve code readability.